### PR TITLE
Prem redux

### PR DESCRIPTION
**Pull request by jim-is-me**

### CODE REVIEW

**Comment by jim-is-me** [comment]
error: failed to push some refs to 'https://github.com/jim-is-me/Route-gaurd.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Comment by CodeNinja47** [comment]
It is a good practice to move all API calls into Redux. Trigger the API calls using actions from inside the `HomeWithRedux.js` component.

File: src/routegaurded_pages/HomeWithRedux.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        setEdit(todo);
+    }
+
+    const saveTodo = async (todo) => {
+        if(todo === undefined){
+            alert("ENter a value")
+            return
+        }
+        if(todo._id !== undefined){
+            const result = await axios.put(`http://localhost:8000/todos/${todo._id}`,{
```
Comment (CodeNinja47) [inline src/routegaurded_pages/HomeWithRedux.js:26]: Move this APi call into redux

File: src/routegaurded_pages/HomeWithRedux.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                data: todo,
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
+                }
+            }).then(resp=>{
+                console.log(resp)
+                alert("Saved")
+            })
+        } else {
+            const result = await axios.post(`http://localhost:8000/todos/`,{
```
Comment (CodeNinja47) [inline src/routegaurded_pages/HomeWithRedux.js:36]: Move this APi call into redux

File: src/routegaurded_pages/HomeWithRedux.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                data: todo,
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
+                }
+            }).then(resp=>{
+                console.log(resp)
```
Comment (CodeNinja47) [inline src/routegaurded_pages/HomeWithRedux.js:42]: Pass a callback function to the redux action. Trigger the callback function once the API call end. Ask me if you don't understand how

File: src/slices/appSlice.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            const data = await result.json();
+            dispatch(setTodos(data));
+            return data;
+        } catch (e) {
+            dispatch(setTodos([]));
+            return '';
+        }
+    };
+};
+
```
Comment (CodeNinja47) [inline src/slices/appSlice.js:30]: Write 2 more actions here to `addTodo` and `updateTodo` which will call the respective APIs.

**Review by CodeNinja47 — CHANGES_REQUESTED** [review]

**Pull request merged**